src/lib.rs

```diff
@@ -48,25 +44,36 @@ impl LinkDrop {
     }
 
     /// Create new account and and claim tokens to it.
-    pub fn create_account_and_claim(&mut self, new_account_id: AccountId, new_public_key: String) -> Promise {
-        assert_eq!(env::signer_account_id(), env::current_account_id());
+    pub fn create_account_and_claim(&mut self, new_account_id: AccountId, new_public_key: Base58PublicKey) -> Promise {
+        assert_eq!(env::signer_account_id(), env::current_account_id(), "Create account and claim only can come from this account");
```

Comment:
It's always safer to use `predecessor_account_id` instead of `signer_account_id`

Reply:
Weird to use `predecessor_account_id` here because it should be signer key for this account.

Reply:
In general contracts should not rely on `signer_account_id`, because if the contract allows to do a promise to some other contract, the other contract can call you back at any method with any input. If it's relies on the `signer_account_id`, then it would be assume the method was called directly.

In your case it doesn't matter, but it usually never a good idea to rely on the `signer_account_id` unless you need to identify the source.

Maybe in what you're trying to do is to check that both `env::signer_account_id()` and `predecessor_account_id()` are the same.

Reply:
The check of `predecessor_account_id() == current_account_id ()` proves that you called this method directly.
`signer_account_id` doesn't give this guarantee